libgamestream/client.c

```diff
@@ -583,10 +603,10 @@ int gs_start_app(PSERVER_DATA server, STREAM_CONFIGURATION *config, int appId, b
   else
     goto cleanup;
 
-  if ((ret = xml_search(data->memory, data->size, "cancel", &result)) != GS_OK)
+  if ((ret = xml_search(data->memory, data->size, "gamesession", &result)) != GS_OK)
     goto cleanup;
 
-  if (atoi(result) == 0) {
+  if (!strcmp(result, "0")) {
```

Comment:
I'm using strcmp() instead of atoi() because "gamesession" is actually a 128-bit hex number in modern GFE versions rather than a 32-bit decimal number like it used to be.